### PR TITLE
Remove asid from vm_map and pmap interface

### PR DESCRIPTION
**Pull request by coodie**
While working on changing vm_map for user threads i realised it is better to keep asids away from interface. In future, when we have processes we might want to create asid or pid allocator, but for now simple incrementation until we run out of asids should be enough.

### CODE REVIEW

**Comment by cahirwpz** [comment]
Please remember to run `clang-format` on your changes :)

File: mips/pmap.c [hunk 1]
```diff
@@ -23,7 +23,6 @@
 #define PTF_SIZE (PTF_ENTRIES * sizeof(pte_t))
 #define PT_ENTRIES (PD_ENTRIES * PTF_ENTRIES)
 #define PT_SIZE (PT_ENTRIES * sizeof(pte_t))
-#define MAX_ASID 255
```
Comment (coodie) [inline mips/pmap.c]: No it's not. Asids are platform dependent, as far as I know some MIPS architectures have different max asids. I couldn't spot max asid macro in toolchain so i decided it's best to use ENTRY_HI mask.

**Comment by rafalcieslak** [comment]
This is great! I will need this change for experimenting with loading multiple user programs.

**Comment by coodie** [comment]
Running clang-format on mips/pmap.c pollutes this file with other formatting changes not associated with my change, because clang-format wasn't used on this file everytime it was changed.

**Pull request merged**